MetaMask/metamask-mobile
# [e2e] Add Ganache seeder for deploying contracts and add simple Send ETH test to Multisig contract

**Description**
- This PR adds the ability to easily deploly smart contracts on the background using Ganache
- The smart contracts directly available are the ones provided by the `test-dapp` dependency
- As a simple showcast, this PR also adds a simple e2e test where we require a Multisig contract to be deployed on Ganache, for later sending ETH to that account
- The test implementation is done with Appium targetting Android, but the Ganache seeder files can be used both by iOS and Android
- At the moment the Multisig contract address is hardcoded on the test. Pending to resolve how to grab variables from the global/background scope into the `.features` file. This can be done in a subsequent PR

**Screenshots/Recordings**

![Screenshot from 2023-04-20 12-18-07](https://user-images.githubusercontent.com/54408225/233337283-9c52b33d-17ce-4188-ae28-b8da660670bf.png)

[Screencast from 20-04-23 12:16:29.webm](https://user-images.githubusercontent.com/54408225/233337319-c0b2b6b6-3b68-4908-bd7a-16e63c57f4a8.webm)


**Manual Testing Steps**
You can check the successful Bitrise job [here](https://app.bitrise.io/build/fac73069-2fc5-49fa-a985-2da7c564a130)

![image](https://github.com/MetaMask/metamask-mobile/assets/54408225/c379f09c-a9fc-4324-a547-0aa102e8d281)

For testing this locally you can do the following:

- Connect an Android device to your computer / use simulator (see Android setup on readme)
- You'll need to set the following vars
```
export ANDROID_HOME=/home/PATH_TO_ANDROID/Android/Sdk
export ANDROID_SDK_ROOT=/home/PATH_TO_ANDROID/Android/Sdk
```

- Open a terminal
- Create the QA build
  - `yarn setup`
 - `yarn start:android:qa`
- Open another terminal
- run `yarn watch`
- On the first terminal run the confirmations suite using this command `yarn test:wdio:android --suite confirmations`

**Screenshots/Recordings**
/

**Issue**
- Resolves https://github.com/MetaMask/mobile-planning/issues/897

**Checklist**
